turbopack/crates/turbopack-core/src/issue/mod.rs

```diff
@@ -518,6 +545,71 @@ impl IssueSource {
     }
 }
 
+async fn source_pos(
+    source: Vc<Box<dyn Source>>,
+    origin: Vc<FileSystemPath>,
+    start: SourcePos,
+    end: SourcePos,
+) -> Result<Option<(Vc<Box<dyn Source>>, SourcePos, SourcePos)>> {
+    let Some(generator) = Vc::try_resolve_sidecast::<Box<dyn GenerateSourceMap>>(source).await?
+    else {
+        return Ok(None);
+    };
+
+    let srcmap = generator.generate_source_map();
+
+    let Some(srcmap) = *convert_to_turbopack_source_map(srcmap, origin).await? else {
+        return Ok(None);
+    };
+
+    let find = |line: usize, col: usize| async move {
+        let TokenWithSource {
+            token,
+            source_content,
+        } = &*srcmap.lookup_token_and_source(line, col).await?;
+
+        match &*token.await? {
+            crate::source_map::Token::Synthetic(t) => Ok::<_, anyhow::Error>((
+                SourcePos {
+                    line: t.generated_line as _,
+                    column: t.generated_column as _,
+                },
+                *source_content,
+            )),
+            crate::source_map::Token::Original(t) => Ok((
+                SourcePos {
+                    line: t.original_line as _,
+                    column: t.original_column as _,
+                },
+                *source_content,
+            )),
+        }
+    };
+
+    let (start, content_1) = find(start.line, start.column).await?;
+    let (end, content_2) = find(end.line, end.column).await?;
+
+    let (content_1, content_2) = match (content_1, content_2) {
+        (Some(content_1), Some(content_2)) => (content_1, content_2),
+        _ => return Ok(None),
+    };
+    let content_1 = content_1.await?;
+    let content_2 = content_2.await?;
+
+    if content_1 != content_2 {
+        return Ok(None);
+    }
+
+    let file = File::from(content_1.clone_value());
+
+    let content = FileContent::new(file).cell();
+    let content = AssetContent::file(content);
+
+    let source = Vc::upcast(VirtualSource::new(source.ident().path(), content));
```

Comment:
`source.ident().path()` is not always correct here. You need to return the name of the source from `lookup_token_and_source` too

Reply:
Maybe return a `Vc<Box<dyn Source>>` from `lookup_token_and_source` so it covers all of that together